src/flow/darcy_flow_mh.cc

```diff
@@ -184,7 +222,7 @@ DarcyFlowMH::EqData::EqData()
     	water_source_density.units( UnitSI().s(-1) );
     
     ADD_FIELD(bc_type,"Boundary condition type, possible values:", "\"none\"" );
-        bc_type.input_selection(&bc_type_selection);
+        bc_type.input_selection( &get_bc_type_selection() );
```

Comment:
Mark this as UGLY, we should try to get rid of pointer to the selection after having generic types.